tools/test-proxy/Azure.Sdk.Tools.TestProxy/Store/GitProcessHandler.cs

```diff
@@ -0,0 +1,168 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Azure.Sdk.Tools.TestProxy.Common;
+using Azure.Sdk.Tools.TestProxy.Common.Exceptions;
+
+namespace Azure.Sdk.Tools.TestProxy.Store
+{
+
+    public class CommandResult
+    {
+        public int ExitCode;
+        public string StdErr;
+        public string StdOut;
+        public string Arguments;
+        public Exception CommandException;
+    }
+
+    /// <summary>
+    /// This class offers an easy wrapper abstraction for shelling out to git.
+    /// </summary>
+    public class GitProcessHandler
+    {
+        public virtual ProcessStartInfo CreateGitProcessInfo(GitAssetsConfiguration config)
+        {
+            var startInfo = new ProcessStartInfo("git")
```

Comment:
Looks like you chose to shell out instead of us the git library. For my knowledge was there a particular reason for this choice? Did the native git library dependency cause a challenge?

Reply:
`sparse checkout` is not supported. Even barring that, it's quite different from using traditional git. (EG have to find their way to invoke commands) If I'm going to climb through a mess of exceptions I'll do it against something that the team has prior knowledge on.

Reply:
@weshaggard we have also been discussing making this class and the `CommandHandler` class common packages. I'm interesting in using test-proxy as a starting point to compare the shelling out experience vs. powershell.